File: CS50 intro/Pset/pset6/mario/mario.py
from cs50 import get_int


def main():
    # Ask for height between 1 and 8 inclusive
    height = get_integer_between_1_and_8_inclusive()
    # Draw left and right aligned pyramids
    draw_pyramids(height)


def draw_pyramids(h):
    row = h - 1
    space = h - 1
    # For pyramids' height from top to bottom (row by row):
    for i in range(h):
        # Right align pyramid
        for k in range(space, 0, -1):
            print(" ", end="")
        # Print left pyramid's blocks
        for j in range(row, h):
            print("#", end="")
        # Print space between left and right pyramids
        print("  ", end="")
        # Print right pyramid's blocks
        for l in range(row, h):
            print("#", end="")
        # Change line aka row
        print("")
        row -= 1
        space -= 1
        

def get_integer_between_1_and_8_inclusive():
    while True:
        n = get_int("Height : ")
        if n < 9 and n > 0:
            break
    return n


main()

File: CS50 intro/Pset/pset1/mario/mario.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    // Height of pyramid
    
    int height;
    do
    {
        height = get_int("Height : ");
    }
    while (height < 1 || height > 8);
    
    //Print righ aligned pyramid
    
    int row = height - 1;
    int space = height - 1;
    
    for (int i = 0; i < height; i++)  
    {
        for (int k = space; k > 0; k--) // Loop to right align pyramid
        {    
            printf(" ");
        }
        for (int j = row; j < height; j++) // Loop to create block #
        {
            printf("#");
        }
        printf("  ");
        
        for (int l = row; l < height; l++)
        {
            printf("#");
        }
        printf("\n");
        row --;
        space--;
    }
}
